pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -0,0 +1,343 @@
+		if err = utils.AddConfigMapWatch(c, configMapName, common.CalicoNamespace); err != nil {
+			return fmt.Errorf("applicationlayer-controller failed to watch ConfigMap %s: %v", configMapName, err)
+		}
+	}
+
+	// Watch for changes to FelixConfiguration.
+	err = c.Watch(&source.Kind{Type: &crdv1.FelixConfiguration{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("applicationlayer-controller failed to watch FelixConfiguration resource: %w", err)
+	}
+
+	return nil
+}
+
+// Blank assignment to verify that ReconcileCompliance implements reconcile.Reconciler.
+var _ reconcile.Reconciler = &ReconcileApplicationLayer{}
+
+// ReconcileApplicationLayer reconciles a ApplicationLayer object.
+type ReconcileApplicationLayer struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver.
+	client          client.Client
+	scheme          *runtime.Scheme
+	provider        operatorv1.Provider
+	status          status.StatusManager
+	clusterDomain   string
+	licenseAPIReady *utils.ReadyFlag
+}
+
+// Reconcile reads that state of the cluster for a ApplicationLayer object and makes changes
+// based on the state read and what is in the ApplicationLayer.Spec.
+func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ApplicationLayer")
+
+	applicationLayer, err := getApplicationLayer(ctx, r.client)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			reqLogger.Info("ApplicationLayer object not found")
+			r.status.OnCRNotFound()
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Error querying for Application Layer")
+		r.status.SetDegraded("Error querying for Application Layer", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	preDefaultPatchFrom := client.MergeFrom(applicationLayer.DeepCopy())
+
+	updateApplicationLayerWithDefaults(applicationLayer)
+
+	err = validateApplicationLayer(applicationLayer)
+
+	if err != nil {
+		reqLogger.Error(err, err.Error())
+		r.status.SetDegraded(err.Error(), "")
+		return reconcile.Result{}, nil
+	}
+
+	// Write the application layer back to the datastore, so the controllers depending on this can reconcile.
+	if err = r.client.Patch(ctx, applicationLayer, preDefaultPatchFrom); err != nil {
+		reqLogger.Error(err, "Failed to write defaults to applicationLayer")
+		r.status.SetDegraded("Failed to write defaults to applicationLayer", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	variant, installation, err := utils.GetInstallation(ctx, r.client)
```

Comment:
We should be checking that variant is TigeraSecureEnterprise like https://github.com/tigera/operator/blob/master/pkg/controller/authentication/authentication_controller.go#L175